### DISCUSSION TITLE
Permission error when starting agate as a service

### DISCUSSION THREAD

**Discussion by gblackmore** [mbrubeck/agate#235] [Q&A]
Hi

When starting agate under the gemini user, I get a permission error while trying to generate certificates for the domain.

Has anyone else run into this problem?

Thanks in advance

edit:

the line in agate.service used to start the server is:

ExecStart=/var/gemini/bin/agate --content /var/gemini/gemini --hostname <fqdn> --addr [::]:1965 --addr 0.0.0.0:1965 --lang en-GB

**Comment by Johann150** [comment]
Since you did not specify the `--certs` flag, agate will try to create the certificates in the current working directory. Please make sure whatever user is used to run the service has write permissions there, or provide the `--certs` option to specify where the certificates should be stored.

You can also start agate with another user once to generate the certificates, after that it will only need to be able to read the certificates from the respective place (provide `--certs` if necessary of course).
